python/lsst/daf/butler/registry/_registry.py

```diff
@@ -255,31 +199,30 @@ def fromConfig(cls, config: Union[ButlerConfig, RegistryConfig, Config, str], cr
         opaque = doImport(config["managers", "opaque"])
         dimensions = doImport(config["managers", "dimensions"])
         collections = doImport(config["managers", "collections"])
+        datasets = doImport(config["managers", "datasets"])
         return cls(database, universe, dimensions=dimensions, opaque=opaque, collections=collections,
-                   create=create)
+                   datasets=datasets, create=create)
 
     def __init__(self, database: Database, universe: DimensionUniverse, *,
                  opaque: Type[OpaqueTableStorageManager],
                  dimensions: Type[DimensionRecordStorageManager],
                  collections: Type[CollectionManager],
+                 datasets: Type[DatasetRecordStorageManager],
                  create: bool = False):
         self._db = database
         self.storageClasses = StorageClassFactory()
         with self._db.declareStaticTables(create=create) as context:
             self._dimensions = dimensions.initialize(self._db, context, universe=universe)
             self._collections = collections.initialize(self._db, context)
-            self._tables = context.addTableTuple(makeRegistryTableSpecs(self.dimensions, self._collections))
+            self._datasets = datasets.initialize(self._db, context,
+                                                 collections=self._collections,
+                                                 universe=self._dimensions.universe)
+            self._tables = context.addTableTuple(makeRegistryTableSpecs(self.dimensions,
```

Comment:
<sigh>.  No real reason; I'll be more consistent within this method at least.  I wish I had named `self.dimensions` -> `self.universe` way back when, but that'd be pretty disruptive now, and maybe that that wish reflects an internal-to-registry rather than user-of-registry perspective anyway.